sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Listeners/PollLogsStrategy.cs

```diff
@@ -78,8 +81,19 @@ public async Task RegisterAsync(BlobServiceClient blobServiceClient, BlobContain
 
             if (!_logListeners.ContainsKey(blobServiceClient))
             {
-                BlobLogListener logListener = await BlobLogListener.CreateAsync(blobServiceClient, _logger, cancellationToken).ConfigureAwait(false);
-                _logListeners.Add(blobServiceClient, logListener);
+                try
+                {
+                    BlobLogListener logListener = await BlobLogListener.CreateAsync(blobServiceClient, _logger, cancellationToken).ConfigureAwait(false);
+                    _logListeners.Add(blobServiceClient, logListener);
+                }
+                catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.AuthorizationPermissionMismatch ||
+                                                        ex.ErrorCode == BlobErrorCode.InsufficientAccountPermissions ||
+                                                        ex.ErrorCode == BlobErrorCode.AuthorizationFailure)
+                {
+                    // Log which account we couldn't enable logging on due to permissions. Ensure that no SAS token is logged on the Uri.
+                    BlobUriBuilder blobUriBuilder = new BlobUriBuilder(blobServiceClient.Uri){ Sas = null };
```

Comment:
Rather than logging the URI, can't you just log the blobServiceClient.AccountName ? Seems the safest and gives you what you need.